### ISSUE TITLE
2-ва СВД-15 и 2-ва СВТ-15 докодить интеграцию до прома.

### ISSUE THREAD

**Issue by Keks1975** [Zud71/elehant_meter#15]
Привет!
Имеется 2-ва СВД-15 и 2-ва СВТ-15
Интеграция обнаружила счётчики, кроме одного СВТ-15, не стал разбираться. Тестил.
Счетчики СВД-15 подтянули сенсоры и показания.
Счетчик СВТ-15 с одним ID обнаружил дважды, с верными показаниями холодной воды и горячей, но как два устройства. с одинаковыми сенсорами, то есть WI-Fi, Батарея и температура были одинаковые для хол и гор воды.
Со временем рэндомно счетчики становились недоступны, снес интеграцию. 
Если нужно могу установить и если потребуется дать логи и показания)))).

**Comment by Zud71** [comment]
Здравствуйте!
Тут ситуация такая, насколько я понимаю работу счетчика. Счетчик по своему усмотрению шлет данные и в этот момент ваше блютуз устройство должно находится в режиме "сканирования", а дальше многое зависит от того как вы настроили блютуз устройство т.к. они должны попасть в окно передачи/сканирования.
Второй момент, как далеко находится ваше устройство т.к. блютуз работает на короткие дистанции.
Счетчик СВТ-15 двухтарифный и он работает как буд-то у вас два устройства, так сделал производитель. Этот момент можно как-то обыграть, но это надо смотреть.
Включите логи в конфиге /config/configuration.yaml
```
logger:
  default: warning
  logs:
    homeassistant.components.bluetooth: debug
    custom_components.elehant_meter: debug
```

И пусть поработает денек, и потом сюда прикрепите. И данные по счетчикам напишите, кто есть кто т.е. их маки и модель и какой проблемный, какой как два отображается ...

**Comment by Keks1975** [comment]
Добрый день!
Коплю логи и маки, куда отправлять? не хотелось бы выкладывать свое нижнее белье в эфир.))))
Между делом, счетчики рэндомна уходят в режим недоступен, 

Дано:
Блютуз интегрирован в мат плату, не свисток. На данный момент работает интеграция "[Elehant Water](https://github.com/SzenProgs/elehant_water)" в штатном  режиме и без нареканий, но с предупреждением от НА
Регистратор: homeassistant.const
Источник: helpers/deprecation.py:222
Первое сообщение: 10:00:10 (3 сообщений)
Последнее сообщение: 10:00:10

VOLUME_LITERS was used from elehant_water, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfVolume.LITERS instead, please report it to the author of the 'elehant_water' custom integration
VOLUME_CUBIC_METERS was used from elehant_water, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfVolume.CUBIC_METERS instead, please report it to the author of the 'elehant_water' custom integration
TEMP_CELSIUS was used from elehant_water, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please report it to the author of the 'elehant_water' custom integratio  
в нынешней интеграции не хватает отсутствие заряда батарейки и сигнала + устаревшая константа, которая будет удалена в HA Core 2025.1. и самое главное проект МЕРТВ. 

Два счетчика СВД-15 и два СВТ-15:
Elehant meter
**Счетчики на Кухне**
ГВ Кухня - 40 СВТ-15: 0006394
1 устройство и 4 объекта

ГВ Кухня + 40 СВТ-15: 0006394
1 устройство и 4 объекта

ХВ Кухня СВД-15: 0082091
1 устройство и 4 объекта

**Счетчики в Санузеле**
ГВ Санузел - 40 СВТ-15: 0006407
1 устройство и 4 объекта

ГВ Санузел + 40 СВТ-15: 0006407
1 устройство и 4 объекта

ХВ Санузел СВД-15: 0082156
1 устройство и 4 объекта

Прикладываю скрины.
![elehant_meter 2](https://github.com/user-attachments/assets/1acddd67-64a6-4c0a-af2c-dfbe01664840)
![elehant_meter 3](https://github.com/user-attachments/assets/7586221e-e6c9-418e-bfe5-0be2ea67bf4a)
![elehant_meter](https://github.com/user-attachments/assets/30358ac8-e9f8-4028-9b72-a32622afed4b)

**Comment by Zud71** [comment]
uadmin@orycan.ru - можете сюда отправить. Спасибо.  Elehant Water работает на прямую с устройством, у нас мы работаем через прослойку. Вам надо бы поиграться с параметрами сканирования устройством в HA ... как это сделать я не знаю. Либо поставить esp32, там более понятно как и где менять параметры.

**Comment by Keks1975** [comment]
Заметил одну штуку, батарея при установки вашей интеграции была 97% сутки поработала интеграция стала  88%, опасаться?
![elehant_meter 4](https://github.com/user-attachments/assets/6c09f15d-980a-4ee3-96fe-0a6a8575c896)

**Comment by Zud71** [comment]
А сколько лет счетчику? В приложение какое состояние батареи? Т.к. я не производитель данных устройств, я мало что могу сказать, является ли такое падение напряжение на батареи опасным или нет. Вам лучше связаться с тех поддержкой. По факту, устройство только передает данные в эфир и никак не общается с блютуз устройством т.е. интеграция никак не влияет на его работу, но это не точно.

**Comment by Keks1975** [comment]
Не использовал esp32 и нашел интересную информацию, может пригодиться. 
статья [https://www.mytechnote.ru/oborudovanie/obzory-ru/bluetooth-shlyuz-dlya-umnogo-doma-za-kopeyki-i-za-5-minut-raboty-na-esp32/](url) 
`Обратите внимание на параметр active:false – есть подозрение, что если его не указывать, то он по умолчанию в состоянии true, и как следствие может негативно влиять на разряд батареи в устройствах`

**Comment by Keks1975** [comment]
Подскажите как снять целиком и полностью логи? пустяк, а не могу снять))))

> А сколько лет счетчику? В приложение какое состояние батареи? Т.к. я не производитель данных устройств, я мало что могу сказать, является ли такое падение напряжение на батареи опасным или нет. Вам лучше связаться с тех поддержкой. По факту, устройство только передает данные в эфир и никак не общается с блютуз устройством т.е. интеграция никак не влияет на его работу, но это не точно.

Установлены в марте 2024, когда произведены не в курсе. 
В приложении просто значок батареи без цифр.

**Comment by Zud71** [comment]
Как  я писал ранее, мы не устанавливаем соединение с устройством и никак не влияем на его батарею, при этом у вас не esp32... Вам надо включить режим дебаг, писал выше, и когда накопятся логи, просто скачать файл home-assistant.log, либо скопировать содержимое ctrl-c  ctrl-v  в другой файл

**Comment by Zud71** [comment]
Попробуйте код из ветки TEST.

**Comment by Keks1975** [comment]
[https://github.com/Zud71/elehant_meter/tree/test](url) Здесь?

**Comment by Keks1975** [comment]
Перегрузил и вот.
![СВТ и СВД](https://github.com/user-attachments/assets/4e58d9cc-ae96-4a80-9930-56394a0a81b2)
логи предоставлю через сутки. 
Добавляю устройство 

![СВТ и СВД 2](https://github.com/user-attachments/assets/0e157373-5105-4864-b95e-d4d45f1850a3)

**Comment by Keks1975** [comment]
Добавляю новый счетчик с  одинаковым номером получаю ошибку 
![СВТ и СВД 3](https://github.com/user-attachments/assets/08e667e9-8b67-4af1-83ff-03a797a6737b)

**Comment by Zud71** [comment]
Не надо добавлять счетчик с одним номером, он не должен быть обнаружен, будем смотреть. Посмотрите в карточки, у вас должно появится горячая и холодная вода под одним номером...

**Comment by Zud71** [comment]
Обновите из ветки TEST  и попробуйте подобавлять устройства. Лога мне хватит сразу после попыток подобавлять, большой лог мне не нужен. 
Устройства с одинаковыми серийными номерами не должны обнаруживаться.
Данные по холодной и горячей воде должны быть в одной карточке, как одно устройство.

**Comment by Keks1975** [comment]
Нет объектов
![СВТ и СВД 4](https://github.com/user-attachments/assets/667318f4-e313-4f27-ab87-ff9d1b3833d5)

**Comment by Zud71** [comment]
Доработка будет не легкой т.к. не могу у себя тестить результат
Допустил глупую ошибку в коде, обновите код. Будем ловить следующий баг

**Comment by Keks1975** [comment]
Крутая работа, я готов тестировать)))
![СВТ и СВД 5](https://github.com/user-attachments/assets/3daa242c-b839-446b-91bf-50a54e559290)
объекты появились, но нет показания горячей воды на СВТ счетчиках.

**Comment by Keks1975** [comment]
Добавляю ручками счетчики видятся и при добавлении ошибка
![СВТ и СВД 6](https://github.com/user-attachments/assets/0e2d7484-804b-4d67-b56e-fa1f8d0057d5)
![СВТ и СВД 7](https://github.com/user-attachments/assets/3d7064c6-8649-451a-a3c4-71d86284ac6c)

**Comment by Zud71** [comment]
Так. Не буду вас больше мучить, на днях тогда эмулятор сделаю для отладки.

**Comment by Keks1975** [comment]
> Так. Не буду вас больше мучить, на днях тогда эмулятор сделаю для отладки.

да без проблем, обращайтесь.

**Comment by Zud71** [comment]
Сделать так, чтоб два устройства виделись как одно у меня не получилось из-за особенностей работы HA .... 
Единственное, что могу это дать разные имена устройствам, что и сделал. 
В ветку TEST загрузил...

**Comment by Zud71** [comment]
![image](https://github.com/user-attachments/assets/aa3a00fb-4b2b-43f3-b531-14180ef2af6c)
![image](https://github.com/user-attachments/assets/88507bdd-97b0-482f-9bc2-0efcd35a6489)

**Comment by Keks1975** [comment]
Привет! Прошу прошение за пропажу.
Полет нормальный, логи присылать?

**Comment by Zud71** [comment]
А что есть проблемы?

**Comment by Keks1975** [comment]
Нет, нету, думал может у вас есть мысль почитать и проанализировать.

**Comment by Zud71** [comment]
Я же сделал себе эмулятор и погонял чуток, там если что-то можно сделать, то это надо лезть куда глубже данного расширения... Если бы производитель не менял маки а передавал двумя пакетами или еще как, то было бы проще. Еще возможно как-то обойти ситуацию, это создание карточки устройства в ручную в конфигурации НА, но ....

**Comment by Keks1975** [comment]
Когда будет новая версия?, сижу на тестовой

**Comment by Zud71** [comment]
Самая последняя версия залита в обе ветки

**Comment by Keks1975** [comment]
v0.1.1 залито 3 месяца назад. или скачать ручками?
![Снимок](https://github.com/user-attachments/assets/7cf4baa7-cbdd-40f7-92ce-e19ab2ca1cd6)

**Comment by Zud71** [comment]
Создал релиз.

**Comment by Keks1975** [comment]
Установил, работает. можно закрывать.

**Comment by sanek119** [comment]
Прошу помощи чтоб настроить счётчики 
Хочу сделать через плату esp32 на прошивке esphome
Сейчас уже установлен газовый, на подходе водяные
Я правильно понимаю что в прошивке esp home надо сделать ссылку на кастом компонент? 
Прошу строго не ругать и если можно прям по порядку подсказать, знаний не хватает (

**Comment by Zud71** [comment]
Установка esp home: https://io-home.ru/home-assistant/integrations/dobavljaem-esphome-v-home-assistant/
Код для ESP и как установить расширение есть тут: https://orycan.ru/blog/post/elehant_meter

**Comment by sanek119** [comment]
Могут написать Вам в телеграм?
--
Отправлено из  Mail для Android пятница, 14 февраля 2025г., 10:59 +04:00 от Yuiri  ***@***.*** :

**Comment by Zud71** [comment]
Все интересующие вопросы вы можете задать здесь. Вам стоит потратить больше времени и чуть чуть изучить тему. Там все достаточно просто.
